src/common/registry.rs

```diff
@@ -22,15 +22,15 @@ lazy_static::lazy_static! {
 
 #[derive(Debug, Eq, PartialEq)]
 pub enum RegistryError {
-    ModelNotFound,
+    ModelNotFound(String),
     ModelAlreadyRegistered(&'static str),
 }
 
 impl fmt::Display for RegistryError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            RegistryError::ModelNotFound => {
-                write!(f, "RegistryError : Model not found")
+            RegistryError::ModelNotFound(model) => {
+                write!(f, "RegistryError : Model '{model}' not found")
             }
             RegistryError::ModelAlreadyRegistered(model) => {
                 write!(f, "RegistryError : model {model} already exists")
```

Comment:
You might be interested in using the [`thiserror`](https://docs.rs/thiserror/latest/thiserror/) crate as an alternative to hand-rolling errors.

Reply:
yeah, it looks like a good option. Thanks for the suggestion!